pinot-spi/src/main/java/org/apache/pinot/spi/utils/TimestampUtils.java

```diff
@@ -34,6 +33,8 @@ public class TimestampUtils {
       .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
       .parseDefaulting(ChronoField.SECOND_OF_MINUTE, 0)
       .toFormatter();
+  private static final DateTimeFormatter ISO8601_FORMATTER =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss[.SSSSSS][.SSSSS][.SSSS][.SSS][.SS][.S]X");
```

Comment:
Do we need all these alternatives? Is there a easier way to define the pattern?

Reply:
Defined in the `UNIVERSAL_DATE_TIME_FORMATTER`